unified-codes/apps/data-service/src/app/schema.ts

```diff
@@ -1,59 +1,98 @@
-import { gql } from 'apollo-server-fastify';
-
-const typeDefs = gql`
-  type Entity {
-    code: String
-    description: String!
-    type: String!
-    uid: String!
-    interactions: [DrugInteraction]
-    children: [Entity]
-    properties: [Property]
-  }
-
-  type Property {
-    type: String!
-    value: String!
-    properties: [Property]
-  }
-
-  type DrugInteraction {
-    name: String
-    description: String
-    severity: String
-    source: String
-    rxcui: String
-  }
-
-  type Query {
-    "Request an entity by code"
-    entity(code: String!): Entity
-    "Request all entities with optional filter - Default behaviour: return all medicinal_products"
-    entities(filter: EntityFilterInput, first: Int, offset: Int): EntityCollection
-  }
-
-  type EntityCollection {
-    data: [Entity]!
-    totalLength: Int!
-  }
-
-  input EntityFilterInput {
-    "Search by entity code"
-    code: String
-    "Search by entity description"
-    description: String
-    "Order search results"
-    orderBy: EntitySortInput
-    "Search by Level in Product Hierarchy"
-    type: String
-  }
-
-  input EntitySortInput {
-    "Field to order by"
-    field: String
-    "Sort ascending"
-    descending: Boolean
-  }
-`;
-
-export { typeDefs };
+import { registerEnumType, ObjectType, InputType, Field, Int } from "type-graphql";
+
+import { IEntity, IEntitySort, IDrugInteraction, IProperty, IEntityCollection, IEntitySearch, EEntityField, EEntityType } from '@unified-codes/data';
+
+// registerEnumType(EEntityType, {
```

Comment:
I'm not super familiar with `type-graphql` yet - does it have some weird enum niggles?

Reply:
Yeah, still figuring it out! Something for another PR, I think!